frontend/src/app/Settings/Users/UserFacilitiesSettingsForm.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useRef, useState, useEffect } from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+import Button from "../../commonComponents/Button";
+import { SettingsUser, UserFacilitySetting } from "./ManageUsersContainer";
+import "./ManageUsers.scss";
+
+interface Props {
+  activeUser: SettingsUser; // the user you are currently attempting to edit
+  allFacilities: UserFacilitySetting[];
+  onUpdateUser(userId: string, key: string, value: UserFacilitySetting[]): void;
+}
+
+const UserFacilitiesSettingsForm: React.FC<Props> = ({
+  activeUser,
+  allFacilities,
+  onUpdateUser,
+}) => {
+  const [isComponentVisible, setIsComponentVisible] = useState(true);
+  const ref = useRef() as React.MutableRefObject<HTMLDivElement>;
+
+  const handleClickOutside = (event: any) => {
```

Comment:
Clicking outside the component makes it go away? We don't do that for other modal things do we? If this isn't modal then clicking to other stuff should be handled by the router I'd think.

Reply:
It's more like a custom dropdown that looks like a modal:

![facilities0 1](https://user-images.githubusercontent.com/35268641/108103750-3cc92d80-703f-11eb-9204-d95b3df19df1.gif)

Reply:
arguably, an actual dropdown would be more...typical

Reply:
OK, I see what you're doing now. An actual dropdown would create fewer accessibility hassles for us as developers, but the native select doesn't have an intuitive way to "select all". At minimum I think the dropdown that comes up needs something like an `aria-alert="true"` but there may be other things that need to be added.